File: 9/python/sample.py
from time import sleep

class sample:
    def end(self, value):
        if value > 0:
            print(value)
            return self.end(value-1)
        print("End")
        return value

    def start(self, value):
        while True:
           print("Start")
           self.end(value)
           sleep(1)

if __name__ == '__main__':
    sample().start(3)

File: 9/ruby/sample.rb
class Sample
	def _end(value)
		if value > 0
			puts(value)
			return _end(value-1)
		end
		puts("end")
		return value
	end

	def _start(value)
		puts("start")
		while true
			_end(value)
			sleep(1)
		end
	end
end

sample = Sample.new()
sample._start(3)
